torchcam/cams/cam.py

```diff
@@ -25,11 +25,16 @@ def __init__(
         conv_layer: str
     ) -> None:
 
-        if not hasattr(model, conv_layer):
+        # Obtain a mapping from module name to module instance for each layer in the model
+        self.model_module_dict = {}
+        for name, module in model.named_modules():
+            self.model_module_dict[name] = module
```

Comment:
Would you mind renaming the attribute to `submodule_dict` since the extract is meant for any Module (not necessarily considered as model)?

Also, since `model.named_modules()` is a generator, you can directly do `dict(model.named_modules())`

Reply:
Sure thing